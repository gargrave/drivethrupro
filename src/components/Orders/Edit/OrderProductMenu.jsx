import React from 'react'
import { array, func, number, shape } from 'prop-types'

const OrderProductMenu = props => {
  return (
    <span className="input-group order-product-menu">
      <label htmlFor="product">Product:</label>
      <select
        name="product"
        id="product"
        className="order-product-select"
        value={props.selectedMenuItem.id}
        onChange={props.handleItemChange}
      >
        {props.menu.map(item => {
          return (
            <option key={item.id} value={item.id} className="order-product-select-option">
              {item.name}
            </option>
          )
        })}
      </select>
    </span>
  )
}

OrderProductMenu.propTypes = {
  menu: array.isRequired,
  selectedMenuItem: shape({
    id: number.isRequired
  }).isRequired,
  handleItemChange: func.isRequired
}

export default OrderProductMenu
