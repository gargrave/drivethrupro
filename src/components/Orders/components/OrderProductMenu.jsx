import React from 'react'
import { array, func, number, shape } from 'prop-types'

const OrderProductMenu = props => {
  return (
    <span className="input-group">
      <label htmlFor="product">Product: </label>
      <select name="product" id="product" value={props.selectedMenuItem.id} onChange={props.handleItemChange}>
        {props.menu.map(item => {
          return (
            <option key={item.id} value={item.id}>
              {item.name}
            </option>
          )
        })}
      </select>
    </span>
  )
}

OrderProductMenu.propTypes = {
  menu: array.isRequired,
  selectedMenuItem: shape({
    id: number.isRequired
  }).isRequired,
  handleItemChange: func.isRequired
}

export default OrderProductMenu
