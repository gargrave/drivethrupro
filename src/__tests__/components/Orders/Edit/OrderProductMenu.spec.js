import React from 'react'
import { shallow } from 'enzyme'

import OrderProductMenu from '../../../../components/Orders/Edit/OrderProductMenu'

import menu from '../../../../data/menu-items'

describe('OrderProductMenu', () => {
  let props

  beforeEach(() => {
    props = {
      menu,
      selectedMenuItem: {
        id: 1
      },
      handleItemChange: jest.fn()
    }
  })

  test('renders correctly', () => {
    const component = shallow(<OrderProductMenu {...props} />)
    expect(component).toMatchSnapshot()
  })

  test('pre-selects the option passed into props', () => {
    const component = shallow(<OrderProductMenu {...props} />)
    const select = component.find('.order-product-select')
    const expected = props.selectedMenuItem.id
    const actual = select.props().value
    expect(actual).toEqual(expected)
  })

  test('renders an <option> for each menu item', () => {
    const component = shallow(<OrderProductMenu {...props} />)
    const expected = menu.length
    const actual = component.find('.order-product-select-option').length
    expect(actual).toEqual(expected)
  })

  test('calls the "change" callback when an option is selected', () => {
    const component = shallow(<OrderProductMenu {...props} />)
    expect(props.handleItemChange.mock.calls.length).toEqual(0)
    component.find('.order-product-select').simulate('change')
    expect(props.handleItemChange.mock.calls.length).toEqual(1)
  })
})
