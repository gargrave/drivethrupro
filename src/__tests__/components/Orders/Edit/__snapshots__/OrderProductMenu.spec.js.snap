// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderProductMenu renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <span
    className="input-group order-product-menu"
>
    <label
        htmlFor="product"
    >
        Product:
    </label>
    <select
        className="order-product-select"
        id="product"
        name="product"
        onChange={[Function]}
        value={1}
    >
        <option
            className="order-product-select-option"
            value={0}
        >
            Hamburger
        </option>
        <option
            className="order-product-select-option"
            value={1}
        >
            Cheeseburger
        </option>
        <option
            className="order-product-select-option"
            value={2}
        >
            Double Cheeseburger
        </option>
        <option
            className="order-product-select-option"
            value={3}
        >
            SUPER ULTRA Burger
        </option>
        <option
            className="order-product-select-option"
            value={4}
        >
            Chicken Sandwich
        </option>
        <option
            className="order-product-select-option"
            value={5}
        >
            French Fries
        </option>
        <option
            className="order-product-select-option"
            value={6}
        >
            Currrly Fries
        </option>
        <option
            className="order-product-select-option"
            value={7}
        >
            Fountrain Drink
        </option>
        <option
            className="order-product-select-option"
            value={8}
        >
            Bottled Water
        </option>
        <option
            className="order-product-select-option"
            value={9}
        >
            Milkshake
        </option>
    </select>
</span>,
  "nodes": Array [
    <span
      className="input-group order-product-menu"
>
      <label
            htmlFor="product"
      >
            Product:
      </label>
      <select
            className="order-product-select"
            id="product"
            name="product"
            onChange={[Function]}
            value={1}
      >
            <option
                  className="order-product-select-option"
                  value={0}
            >
                  Hamburger
            </option>
            <option
                  className="order-product-select-option"
                  value={1}
            >
                  Cheeseburger
            </option>
            <option
                  className="order-product-select-option"
                  value={2}
            >
                  Double Cheeseburger
            </option>
            <option
                  className="order-product-select-option"
                  value={3}
            >
                  SUPER ULTRA Burger
            </option>
            <option
                  className="order-product-select-option"
                  value={4}
            >
                  Chicken Sandwich
            </option>
            <option
                  className="order-product-select-option"
                  value={5}
            >
                  French Fries
            </option>
            <option
                  className="order-product-select-option"
                  value={6}
            >
                  Currrly Fries
            </option>
            <option
                  className="order-product-select-option"
                  value={7}
            >
                  Fountrain Drink
            </option>
            <option
                  className="order-product-select-option"
                  value={8}
            >
                  Bottled Water
            </option>
            <option
                  className="order-product-select-option"
                  value={9}
            >
                  Milkshake
            </option>
      </select>
</span>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <OrderProductMenu
        handleItemChange={[Function]}
        menu={
                Array [
                        Object {
                          "id": 0,
                          "name": "Hamburger",
                          "sizes": Array [
                            Object {
                              "name": "Regular",
                              "price": 99,
                            },
                          ],
                        },
                        Object {
                          "id": 1,
                          "name": "Cheeseburger",
                          "sizes": Array [
                            Object {
                              "name": "Regular",
                              "price": 119,
                            },
                          ],
                        },
                        Object {
                          "id": 2,
                          "name": "Double Cheeseburger",
                          "sizes": Array [
                            Object {
                              "name": "Regular",
                              "price": 189,
                            },
                          ],
                        },
                        Object {
                          "id": 3,
                          "name": "SUPER ULTRA Burger",
                          "sizes": Array [
                            Object {
                              "name": "SUPER ULTRA",
                              "price": 399,
                            },
                          ],
                        },
                        Object {
                          "id": 4,
                          "name": "Chicken Sandwich",
                          "sizes": Array [
                            Object {
                              "name": "Regular",
                              "price": 179,
                            },
                          ],
                        },
                        Object {
                          "id": 5,
                          "name": "French Fries",
                          "sizes": Array [
                            Object {
                              "name": "Small",
                              "price": 49,
                            },
                            Object {
                              "name": "Medium",
                              "price": 69,
                            },
                            Object {
                              "name": "Large",
                              "price": 89,
                            },
                          ],
                        },
                        Object {
                          "id": 6,
                          "name": "Currrly Fries",
                          "sizes": Array [
                            Object {
                              "name": "Small",
                              "price": 79,
                            },
                            Object {
                              "name": "Medium",
                              "price": 119,
                            },
                            Object {
                              "name": "Large",
                              "price": 149,
                            },
                          ],
                        },
                        Object {
                          "id": 7,
                          "name": "Fountrain Drink",
                          "sizes": Array [
                            Object {
                              "name": "Small",
                              "price": 59,
                            },
                            Object {
                              "name": "Medium",
                              "price": 79,
                            },
                            Object {
                              "name": "Large",
                              "price": 99,
                            },
                            Object {
                              "name": "Obscene",
                              "price": 139,
                            },
                          ],
                        },
                        Object {
                          "id": 8,
                          "name": "Bottled Water",
                          "sizes": Array [
                            Object {
                              "name": "Regular",
                              "price": 129,
                            },
                          ],
                        },
                        Object {
                          "id": 9,
                          "name": "Milkshake",
                          "sizes": Array [
                            Object {
                              "name": "Small",
                              "price": 119,
                            },
                            Object {
                              "name": "Medium",
                              "price": 149,
                            },
                            Object {
                              "name": "Large",
                              "price": 179,
                            },
                          ],
                        },
                      ]
        }
        selectedMenuItem={
                Object {
                        "id": 1,
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "handleItemChange": [Function],
          "menu": Array [
            Object {
              "id": 0,
              "name": "Hamburger",
              "sizes": Array [
                Object {
                  "name": "Regular",
                  "price": 99,
                },
              ],
            },
            Object {
              "id": 1,
              "name": "Cheeseburger",
              "sizes": Array [
                Object {
                  "name": "Regular",
                  "price": 119,
                },
              ],
            },
            Object {
              "id": 2,
              "name": "Double Cheeseburger",
              "sizes": Array [
                Object {
                  "name": "Regular",
                  "price": 189,
                },
              ],
            },
            Object {
              "id": 3,
              "name": "SUPER ULTRA Burger",
              "sizes": Array [
                Object {
                  "name": "SUPER ULTRA",
                  "price": 399,
                },
              ],
            },
            Object {
              "id": 4,
              "name": "Chicken Sandwich",
              "sizes": Array [
                Object {
                  "name": "Regular",
                  "price": 179,
                },
              ],
            },
            Object {
              "id": 5,
              "name": "French Fries",
              "sizes": Array [
                Object {
                  "name": "Small",
                  "price": 49,
                },
                Object {
                  "name": "Medium",
                  "price": 69,
                },
                Object {
                  "name": "Large",
                  "price": 89,
                },
              ],
            },
            Object {
              "id": 6,
              "name": "Currrly Fries",
              "sizes": Array [
                Object {
                  "name": "Small",
                  "price": 79,
                },
                Object {
                  "name": "Medium",
                  "price": 119,
                },
                Object {
                  "name": "Large",
                  "price": 149,
                },
              ],
            },
            Object {
              "id": 7,
              "name": "Fountrain Drink",
              "sizes": Array [
                Object {
                  "name": "Small",
                  "price": 59,
                },
                Object {
                  "name": "Medium",
                  "price": 79,
                },
                Object {
                  "name": "Large",
                  "price": 99,
                },
                Object {
                  "name": "Obscene",
                  "price": 139,
                },
              ],
            },
            Object {
              "id": 8,
              "name": "Bottled Water",
              "sizes": Array [
                Object {
                  "name": "Regular",
                  "price": 129,
                },
              ],
            },
            Object {
              "id": 9,
              "name": "Milkshake",
              "sizes": Array [
                Object {
                  "name": "Small",
                  "price": 119,
                },
                Object {
                  "name": "Medium",
                  "price": 149,
                },
                Object {
                  "name": "Large",
                  "price": 179,
                },
              ],
            },
          ],
          "selectedMenuItem": Object {
            "id": 1,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <span
          className="input-group order-product-menu"
>
          <label
                    htmlFor="product"
          >
                    Product:
          </label>
          <select
                    className="order-product-select"
                    id="product"
                    name="product"
                    onChange={[Function]}
                    value={1}
          >
                    <option
                              className="order-product-select-option"
                              value={0}
                    >
                              Hamburger
                    </option>
                    <option
                              className="order-product-select-option"
                              value={1}
                    >
                              Cheeseburger
                    </option>
                    <option
                              className="order-product-select-option"
                              value={2}
                    >
                              Double Cheeseburger
                    </option>
                    <option
                              className="order-product-select-option"
                              value={3}
                    >
                              SUPER ULTRA Burger
                    </option>
                    <option
                              className="order-product-select-option"
                              value={4}
                    >
                              Chicken Sandwich
                    </option>
                    <option
                              className="order-product-select-option"
                              value={5}
                    >
                              French Fries
                    </option>
                    <option
                              className="order-product-select-option"
                              value={6}
                    >
                              Currrly Fries
                    </option>
                    <option
                              className="order-product-select-option"
                              value={7}
                    >
                              Fountrain Drink
                    </option>
                    <option
                              className="order-product-select-option"
                              value={8}
                    >
                              Bottled Water
                    </option>
                    <option
                              className="order-product-select-option"
                              value={9}
                    >
                              Milkshake
                    </option>
          </select>
</span>,
        "_debugID": 2,
        "_renderedOutput": <span
          className="input-group order-product-menu"
>
          <label
                    htmlFor="product"
          >
                    Product:
          </label>
          <select
                    className="order-product-select"
                    id="product"
                    name="product"
                    onChange={[Function]}
                    value={1}
          >
                    <option
                              className="order-product-select-option"
                              value={0}
                    >
                              Hamburger
                    </option>
                    <option
                              className="order-product-select-option"
                              value={1}
                    >
                              Cheeseburger
                    </option>
                    <option
                              className="order-product-select-option"
                              value={2}
                    >
                              Double Cheeseburger
                    </option>
                    <option
                              className="order-product-select-option"
                              value={3}
                    >
                              SUPER ULTRA Burger
                    </option>
                    <option
                              className="order-product-select-option"
                              value={4}
                    >
                              Chicken Sandwich
                    </option>
                    <option
                              className="order-product-select-option"
                              value={5}
                    >
                              French Fries
                    </option>
                    <option
                              className="order-product-select-option"
                              value={6}
                    >
                              Currrly Fries
                    </option>
                    <option
                              className="order-product-select-option"
                              value={7}
                    >
                              Fountrain Drink
                    </option>
                    <option
                              className="order-product-select-option"
                              value={8}
                    >
                              Bottled Water
                    </option>
                    <option
                              className="order-product-select-option"
                              value={9}
                    >
                              Milkshake
                    </option>
          </select>
</span>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <OrderProductMenu
    handleItemChange={[Function]}
    menu={
        Array [
            Object {
              "id": 0,
              "name": "Hamburger",
              "sizes": Array [
                Object {
                  "name": "Regular",
                  "price": 99,
                },
              ],
            },
            Object {
              "id": 1,
              "name": "Cheeseburger",
              "sizes": Array [
                Object {
                  "name": "Regular",
                  "price": 119,
                },
              ],
            },
            Object {
              "id": 2,
              "name": "Double Cheeseburger",
              "sizes": Array [
                Object {
                  "name": "Regular",
                  "price": 189,
                },
              ],
            },
            Object {
              "id": 3,
              "name": "SUPER ULTRA Burger",
              "sizes": Array [
                Object {
                  "name": "SUPER ULTRA",
                  "price": 399,
                },
              ],
            },
            Object {
              "id": 4,
              "name": "Chicken Sandwich",
              "sizes": Array [
                Object {
                  "name": "Regular",
                  "price": 179,
                },
              ],
            },
            Object {
              "id": 5,
              "name": "French Fries",
              "sizes": Array [
                Object {
                  "name": "Small",
                  "price": 49,
                },
                Object {
                  "name": "Medium",
                  "price": 69,
                },
                Object {
                  "name": "Large",
                  "price": 89,
                },
              ],
            },
            Object {
              "id": 6,
              "name": "Currrly Fries",
              "sizes": Array [
                Object {
                  "name": "Small",
                  "price": 79,
                },
                Object {
                  "name": "Medium",
                  "price": 119,
                },
                Object {
                  "name": "Large",
                  "price": 149,
                },
              ],
            },
            Object {
              "id": 7,
              "name": "Fountrain Drink",
              "sizes": Array [
                Object {
                  "name": "Small",
                  "price": 59,
                },
                Object {
                  "name": "Medium",
                  "price": 79,
                },
                Object {
                  "name": "Large",
                  "price": 99,
                },
                Object {
                  "name": "Obscene",
                  "price": 139,
                },
              ],
            },
            Object {
              "id": 8,
              "name": "Bottled Water",
              "sizes": Array [
                Object {
                  "name": "Regular",
                  "price": 129,
                },
              ],
            },
            Object {
              "id": 9,
              "name": "Milkshake",
              "sizes": Array [
                Object {
                  "name": "Small",
                  "price": 119,
                },
                Object {
                  "name": "Medium",
                  "price": 149,
                },
                Object {
                  "name": "Large",
                  "price": 179,
                },
              ],
            },
          ]
    }
    selectedMenuItem={
        Object {
            "id": 1,
          }
    }
/>,
}
`;
